### DISCUSSION TITLE
Announcing the `1.x.x` stable branch

### DISCUSSION THREAD

**Discussion by mrcjkb** [mrcjkb/haskell-tools.nvim#119] [Announcements]
To make it easier to keep up to date while avoiding breaking changes, I have decided to maintain a `1.x.x` stable branch.

Some examples of how install this plugin using the stable branch,:

## packer

```lua
use {
  'mrcjkb/haskell-tools.nvim',
  requires = {
    'nvim-lua/plenary.nvim',
  },
  branch = '1.x.x', 
}
```

## lazy

```lua
{
  'mrcjkb/haskell-tools.nvim',
  dependencies = {
    'nvim-lua/plenary.nvim',
  },
  branch = '1.x.x', 
},
```

I will probably wait a few days before merging feature releases from `master` into `1.x.x`.

If you have any questions or feedback, please don't hesitate to post them here!

Update:

- Removed  'neovim/nvim-lspconfig' from dependencies in code snippets, because this plugin no longer depends on it.

**Comment by jacksonneal** [comment]
Hello! Thank you for the great work on the plugin!  I am working to get my neovim setup running for haskell - new to both and had a few questions/comments.

1. I had previously setup nvim-lspconfig with hls and could not figure out how to stop that from running so that this plugin would work.  Even when removing the hls setup, I could see hls was still running and I would see double virtual text.  The key for me was to run `LspUninstall hls`.  Wanted to share in case its worth adding to the documentation to save another beginner time when migrating.

2. You list `neovim/nvim-lspconfig` as a dependency here but not in the README.md Installation snippet.  Is it needed and how could I verify that it is?

3. I am using LazyVim and cannot seem to get the README.md Quick Setup script loading properly.  Do you have an example of where that should live in my config?

**Comment by mrcjkb** [comment]
Hey 😄 

1. `LspUninstall` sounds like a command from [`mason.nvim`](https://github.com/williamboman/mason.nvim), which I guess is assuming that you are using `nvim-lspconfig` to start the LSP client. I will open an issue to investigate this. Perhaps I can patch `mason.nvim` to detect if you are using `haskell-tools` and use it instead. This might also solve your third issue.

2. At the time of this announcement, `haskell-tools.nvim` did depend on `nvim-lspconfig`. That is no longer the case.
I will update the snippets accordingly.

3. I don't use LazyVim and unfortunately don't have the capacity to officially support all of the different neovim distros out there. But if you post (a link to) your config here, I can take a look and see if I can figure out what's wrong.

**Reply by jacksonneal** [reply]
Ahh yes, I had installed with mason.nvim before as well.  Actually just moved the script to ~/.config/nvim/ftplugin/haskell.lua as recommended and it works fine - I was previously trying to execute in in the lazy.nvim setup block.

**Reply by mrcjkb** [reply]
Glad to hear you could resolve it 😄
